applications/vanilla/modules/class.usercommentsmodule.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * User comments module.
+ *
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ * @package Vanilla
+ * @since 2.3
+ */
+
+/**
+ * Renders recently posted comments from a specific user
+ */
+class UserCommentsModule extends Gdn_Module {
+
+    /** @var int Display limit. */
+    public $limit = 10;
+
+    /** @var  int The ID of the user for whom you are showing the comments. */
+    public $userID = null;
+
+    /**
+     *
+     *
+     * @throws Exception
+     */
+    public function __construct($sender) {
+        parent::__construct();
+        $this->_ApplicationFolder = 'vanilla';
+        $this->fireEvent('Init');
+        //If you are being executed from the profile controller, get the UserID.
+        if (strtolower(val('ControllerName', $sender)) === 'profilecontroller') {
+            $this->userID = $sender->User->UserID;
+        }
+    }
+
+    /**
+     * Get the data for the module.
+     *
+     * @param int|bool $limit Override the number of comments to display.
+     */
+    public function getData($limit = false) {
+        if (!$limit) {
+            $limit = $this->limit;
+        }
+
+        $userModel = new UserModel();
+        $this->setData('User', $userModel->getID($this->userID));
+        $commentsModel = new CommentModel();
```

Comment:
You should verify there is a valid value for the `$userID` property before proceeding.  If you can't query a user record or the associated comments, there's no point in proceeding.